autopilot/top_centrality.go

```diff
@@ -0,0 +1,90 @@
+package autopilot
+
+import (
+	"runtime"
+
+	"github.com/btcsuite/btcutil"
+)
+
+// TopCentrality is a simple greedy technique to create connections to nodes
+// with the top betweenness centrality value. This algorithm is usually
+// referred to as TopK in the literature. The idea is that by opening channels
+// to nodes with top betweenness centrality we also increase our own betweenness
+// centrality (given we already have at least one channel, or create at least
+// two new channels).
+// A different and much better approach is instead of selecting nodes with top
+// centrality value, we extend the graph in a loop by inserting a new non
+// existing edge and recalculate the betweenness centrality of each node. This
+// technique is usually referred to as "greedy" algorithm and gives better
+// results than TopK but is considerably slower too.
+type TopCentrality struct{}
+
+// A compile time assertion to ensure TopCentrality meets the
+// AttachmentHeuristic interface.
+var _ AttachmentHeuristic = (*TopCentrality)(nil)
+
+// NewTopCentrality constructs and returns a new TopCentrality heuristic.
+func NewTopCentrality() *TopCentrality {
+	return &TopCentrality{}
+}
+
+// Name returns the name of the heuristic.
+func (g *TopCentrality) Name() string {
+	return "topk_centrality"
```

Comment:
should we rather name the heuristic simply "centrality"? Feels more right to not have the user have to care about the underlying algorithm, and we can change to the greedy algorithm later without changing the name.

Reply:
Yeah, good point. Done